### PR TITLE
pkp/pkp-lib#8014 swagger stats API documentation

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/8014

### CODE REVIEW

File: docs/dev/swagger-source.json [hunk 1]
```diff
@@ -3799,17 +4370,27 @@
...
 						"name": "dateEnd",
-						"description": "Return stats for publications before this date.",
+						"description": "Return stats before this date.",
 						"in": "query",
 						"required": false,
 						"type": "string"
+					},
+					{
+						"name": "type",
+						"description": "Return issue galley downloads.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4387]: Same change as the `type` for the other issue endpoint.

File: docs/dev/swagger-source.json [hunk 2]
```diff
@@ -3832,7 +4413,883 @@
...
+					}
+				}
+			}
+		},
+		"/stats/publications": {
+			"get": {
+				"tags": [
+					"Stats / Publications"
+				],
+				"summary": "Get a list of publications by total visits.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4436]: > Get a list of publications ordered by visits.

File: docs/dev/swagger-source.json [hunk 3]
```diff
@@ -3832,7 +4413,883 @@
...
+				"description": "Returns a list of publications ordered by their total abstract views. Includes total views by abstract, all galleys, PDF galleys, HTML galleys and other galleys.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4437]: > Returns a list of publications ordered by the number of views of the publication landing page.

File: docs/dev/swagger-source.json [hunk 4]
```diff
@@ -3832,7 +4413,883 @@
...
+				}
+			}
+		},
+		"/stats/publications/timeline": {
+			"get": {
+				"tags": [
+					"Stats / Publications"
+				],
+				"summary": "Get visits by day or month.",
+				"description": "Returns a monthly or daily timeline of abstract views or files downloads for the publications which match the request parameters.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4559]: > Returns a monthly or daily timeline of visits to the publications which match the request parameters.

File: docs/dev/swagger-source.json [hunk 5]
```diff
@@ -3832,7 +4413,883 @@
...
+				"parameters": [
+					{
+						"name": "timelineInterval",
+						"description": "Return the total daily or monthly views",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4563]: > Return the daily or monthly values.

File: docs/dev/swagger-source.json [hunk 6]
```diff
@@ -3832,7 +4413,883 @@
...
+					{
+						"name": "searchPhrase",
+						"description": "Filter results by publications that match this search phrase. Searches the same submission properties as the `searchPhrase` parameter in the `/submissions` endpoint.",
+						"in": "query",
+						"required": false,
+						"type": "string"
+					},
+					{
+						"name": "type",
+						"description": "Return full texts and supplementary files downloads.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4626]: Similar change in the description as other `type` fields.

File: docs/dev/swagger-source.json [hunk 7]
```diff
@@ -3832,7 +4413,883 @@
...
+						"name": "count",
+						"description": "How many publications files to return in a single request. Max is `100`.",
+						"in": "query",
+						"required": false,
+						"type": "integer",
+						"default": 30
+					},
+					{
+						"name": "offset",
+						"description": "Offset the publications files returned. Use this to receive subsequent pages of requests.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4888]: "publications files" -> "files"

Comment (NateWr) [reply docs/dev/swagger-source.json:4888]: Same in other places.

File: docs/dev/swagger-source.json [hunk 8]
```diff
@@ -3832,7 +4413,883 @@
...
+					}
+				}
+			}
+		},
+		"/stats/publications/countries": {
+			"get": {
+				"tags": [
+					"Stats / Publications"
+				],
+				"summary": "Get a list of countries by total publications visits.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4946]: > Get a list of countries ordered by the number of visits to publications.

File: docs/dev/swagger-source.json [hunk 9]
```diff
@@ -3832,7 +4413,883 @@
...
+				"description": "Returns a list of countries ordered by their total publications visits (views and downloads). IP Geolocation is provided by [DB-IP](https://db-ip.com).",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:4947]: > Returns a list of countries ordered by the number of visits to publications from each country. IP Geolocation...

File: docs/dev/swagger-source.json [hunk 10]
```diff
@@ -3832,7 +4413,883 @@
...
+						"enum": [
+							"ASC",
+							"DESC"
+						]
+					}
+				],
+				"responses": {
+					"200": {
+						"x-summary": "Success",
+						"description": "A list of countries with their total publications stats during the requested period.",
```
Comment (NateWr) [inline docs/dev/swagger-source.json:5031]: > A list of countries.

File: docs/dev/swagger-source.json [hunk 11]
```diff
@@ -3832,7 +4413,883 @@
...
+						"x-summary": "Request Not Found",
+						"description": "The requested resource was not found.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					}
+				}
+			}
+		},
+		"/stats/publications/regions": {
```
Comment (NateWr) [inline docs/dev/swagger-source.json:5063]: Same changes as `countries`

File: docs/dev/swagger-source.json [hunk 12]
```diff
@@ -3832,7 +4413,883 @@
...
+						"x-summary": "Request Not Found",
+						"description": "The requested resource was not found.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					}
+				}
+			}
+		},
+		"/stats/publications/cities": {
```
Comment (NateWr) [inline docs/dev/swagger-source.json:5185]: Same changes as `cities`

File: docs/dev/swagger-source.json (hunk 3 above)
Comment (bozana) [reply docs/dev/swagger-source.json:4437]: @NateWr, I think my formulation here was wrong, i.e. somehow 'abstract' sneaked in there :-\ They are ordered by total views which means abstract + all galleys (PDF, HTML and other) views. So maybe:
"Returns a list of publications ordered by the total number of views."?

Comment (NateWr) [reply docs/dev/swagger-source.json:4437]: Good!

**Pull request merged**